### PR TITLE
#167750045 Setup code coverage with coveralls and code-climate

### PR DESCRIPTION
**Pull request by feobaby**
**What does this PR do?**
This PR sets up the tools needed in order to integrate code coverage on coveralls and code-climate.

**Description of tasks to be completed?**
 _install the package dependencies;_
- **coveralls**, in order to get the code coverage
- **nyc**, in order to generate test reports
- **istanbul**, in order to test the coverage of all sections in your code
- **mocha-lcov-reporter**, in order to link mocha to report the coverage data
_then_
- add the repo to the <a href= "https://coveralls.io">**coveralls.io**</a>, in order to get a repo- 
  token and recognize the repo when the unit-tests run
- Add the coverage & codeclimate badges to ReadMe

**How should this be manually tested/checked?**
_After cloning this repo,_ 
- cd into the project
- run the project by typing in some commands;
- run **npm install**, to add all the package dependencies
- _after_, run **npm test** _if you have an endpoint to test already_ and a coverage report will be gotten.

**Any background context you want to provide?**
None for now.

**What are the relevant pivotal tracker stories?**
<a href= "https://www.pivotaltracker.com/story/show/167750045">#167750045 </a>

### CODE REVIEW

**Comment by darasimiolaifa** [comment]
Hi Funmi. Nice work here. But could you please list out the steps needed to test the feature as a list, rather than writing it as a paragraph of steps? That way it is easier to follow as a list of steps and makes it look more orderly. Thanks.

**Comment by feobaby** [comment]
> Hi Funmi. Nice work here. But could you please list out the steps needed to test the feature as a list, rather than writing it as a paragraph of steps? That way it is easier to follow as a list of steps and makes it look more orderly. Thanks.

thanks, I'm on it @darasimiolaifa

**Comment by feobaby** [comment]
> Hi Funmi. Nice work here. But could you please list out the steps needed to test the feature as a list, rather than writing it as a paragraph of steps? That way it is easier to follow as a list of steps and makes it look more orderly. Thanks.

@darasimiolaifa  done!

**Comment by sa-ma** [comment]
@funmi5  Nice job. Some things to note though - your PR description says run test, but there is no test specified

**Review by sa-ma — DISMISSED** [review]

**Comment by feobaby** [comment]
> @funmi5 Nice job. Some things to note though - your PR description says run test, but there is no test specified
@sa-ma yea, that was why I wrote further, that provided there is an **endpoint**.

File: package.json [hunk 1]
```diff
@@ -15,7 +16,9 @@
     "dotenv": "^6.0.0",
     "errorhandler": "^1.5.0",
     "express": "^4.16.3",
+    "istanbul": "^0.4.5",
```
Comment (Oliver-ke) [inline package.json]: Nice one @funmi5 , Could you explain the function of this package, I've not really seen it in use?

**Review by Oliver-ke — DISMISSED** [review]
Hi @funmi5, nice work, While testing this on my local system it generated some error, `Error: ENOENT: no such file or directory, scandir 'G:\projects\wolfsbane-backend\.nyc_output'`, I believe this was due to lack of a test file.

File: package.json (hunk 1 above)
Comment (feobaby) [reply package.json]: oh OK @Oliver-ke  What **_Istanbul_** does is that, it tests all the sections of your code, for instance, if you run **_npm test,_**, **_the report gotten will inform you of the, statements, branches, functions and lines_** covered in the code.

**Comment by feobaby** [comment]
> Hi @funmi5, nice work, While testing this on my local system it generated some error, `Error: ENOENT: no such file or directory, scandir 'G:\projects\wolfsbane-backend\.nyc_output'`, I believe this was due to lack of a test file.

@Oliver-ke  yes, if there is a lack of a test file, and also if no test was ran, since when tests run, it generates a nyc_output file and a coverage report gotten.

**Review by eokwukwe — CHANGES_REQUESTED** [review]
Good job @funmi5. Please every package for test should be in the list of `devDependencies`

**Comment by feobaby** [comment]
> Good job @funmi5. Please every package for test should be in the list of `devDependencies`

@eokwukwe thanks I'm on it.

**Review by torsami77 — DISMISSED** [review]

**Comment by feobaby** [comment]
> Good job @funmi5. Please every package for test should be in the list of `devDependencies`

@eokwukwe Done, I uninstalled the test packages present in the dependencies and installed them back to `devdependencies`

**Review by sa-ma — DISMISSED** [review]

File: package.json [hunk 2]
```diff
@@ -24,10 +25,14 @@
     "@babel/node": "^7.5.5",
     "@babel/preset-env": "^7.5.5",
     "@babel/register": "^7.5.5",
+    "coveralls": "^3.0.6",
     "cross-env": "^5.2.0",
     "eslint": "^5.16.0",
     "eslint-config-airbnb-base": "^13.1.0",
     "eslint-plugin-import": "^2.18.2",
-    "nodemon": "^1.19.1"
+    "istanbul": "^0.4.5",
```
Comment (eokwukwe) [inline package.json]: I don't think we need `instanbul` if we're using `nyc` with `coverallls`. Please confirm this.

Comment (feobaby) [reply package.json]: > I don't think we need `instanbul` if we're using `nyc` with `coverallls`. Please confirm this.

@eokwukwe it's very true, I had to confirm that.

Comment (feobaby) [reply package.json]: @eokwukwe i've made the changes.

File: package.json (hunk 1 above)
Comment (Oliver-ke) [reply package.json]: Alright thanks @funmi5 , we could have also done that using nyc.

Comment (feobaby) [reply package.json]: @Oliver-ke yes I had to uninstall it when I discovered it does the same thing as nyc. Thanks.

**Review by Oliver-ke — DISMISSED** [review]
Good Job @funmi5

**Review by uthdev — DISMISSED** [review]
Nice one @funmi5

**Review by eokwukwe — APPROVED** [review]

**Pull request merged**